src/api/spec/jobs/report_to_scm_job_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
+
+RSpec.describe ReportToScmJob, vcr: false do
+  let(:user) { create(:confirmed_user, login: 'foolano') }
+  let(:token) { Token::Workflow.create(user: user, scm_token: 'fake_token') }
+  let(:project) { create(:project, name: 'project_1', maintainer: user) }
+  let(:package) { create(:package, name: 'package_1', project: project) }
+  let(:repository) { create(:repository, name: 'repository_1', project: project) }
+  let(:event) { Event::BuildSuccess.create({ project: project.name, package: package.name, repository: repository.name, reason: 'foo' }) }
+  let(:event_subscription) do
+    EventSubscription.create(token: token,
+                             user: user,
+                             package: package,
+                             receiver_role: 'reader',
+                             payload: { scm: 'github' },
+                             eventtype: 'Event::BuildSuccess',
+                             channel: :scm)
+  end
+
+  shared_examples 'not reporting to the SCM' do
+    it 'job return value is false' do
+      expect(subject).to be_falsey
+    end
+
+    it 'does not call the scm reporter' do
+      expect_any_instance_of(SCMStatusReporter).not_to receive(:call) # rubocop:disable RSpec/AnyInstance
```

Comment:
This expectation can be refactored later in another PR using a custom status reporter passed as parameter to the `ReportToScmJob`.